src/pages/index.js

```diff
@@ -68,37 +111,115 @@ profileEditButton.addEventListener("click", () => {
   jobInput.value = currentUser.job;
 
   profileModal.open();
-  profileFormValidator.resetValidation();
+  formValidators["edit-profile-form"].resetValidation();
 });
 
 addNewCardButton.addEventListener("click", () => {
   cardModal.open();
 });
 
+profileAvatarButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
 //FUNCTIONS:
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, "#card-template", () => {
-    imagePreviewModal.open(cardData);
-  });
+  const newCard = new Card(
+    cardData, //data
+    "#card-template", //cardSelector
+    () => {
+      //handleImageClick
+      imagePreviewModal.open(cardData);
+    },
+    handleDeleteSubmit,
+    handleLikeClick
+  );
+
   return newCard.generateCard();
 }
 
+function handleSubmit(request, modalInstance, loadingText = "Saving...") {
+  //change button text:
+  modalInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      modalInstance.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      modalInstance.renderLoading(false);
+    });
+}
+
 function handleProfileFormSubmit(inputValues) {
-  console.log(inputValues.name);
-  userInfo.setUserInfo({
-    nameInput: inputValues.name,
-    jobInput: inputValues.job,
-  });
-  profileModal.close();
+  // we create a function that returns a promise
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+    return api
+      .updateUserInfo(inputValues.name, inputValues.job)
+      .then((userData) => {
+        userInfo.setUserInfo({
+          nameInput: inputValues.name,
+          jobInput: inputValues.job,
+        });
+      });
+  }
+  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
+  handleSubmit(makeRequest, profileModal);
 }
 
 function handleAddCardFormSubmit(inputValues) {
-  const name = inputValues.title;
-  const link = inputValues.url;
-  const cardData = { name, link };
+  function makeRequest() {
+    return api.addNewCard(inputValues.title, inputValues.url).then((res) => {
+      cardsSection.prependItem(createCard(res));
+      cardModal.modalForm.reset();
```

Comment:
* You need to **`disable`** the submit button in the `card popup` if the inputs are **empty** when you open it. It’s needed because I can click the button and add an empty card but it should **not** happen. After the first submit the button gets activated and you need to **deactivate** it again. Even if I can't add an empty card the button should look **disabled** (gray) anyway and with attribute `disabled`.

Reply:
when you click the add button, the modal allows you to hit submit with empty input fields? I'm finding it difficult to recreate this on my end.